RealmSwift/ThreadSafeReference.swift

```diff
@@ -122,6 +122,72 @@ public protocol ThreadConfined {
     }
 }
 
+/**
+ Object types conforming to `ThreadSafeWrappable` are able to have the `@ThreadSafe` property wrapper applied to them.
+
+ Note that only types defined by Realm can meaningfully conform to this protocol, and defining new
+ classes which attempt to conform to it will not make them work with `@ThreadSafe`
+ */
+public protocol ThreadSafeWrappable: ThreadConfined {
+    /** The Realm property type associated with this type */
+    static var confinedType: PropertyType { get }
+}
+/**
+    A property type wrapper type that may be passed between threads.
+
+    A `@ThreadSafe` property contains a thread-safe reference to the underlying wrapped value. This reference is resolved to the thread on which the wrapped value is accessed. A new thread safe reference is created each time the property is accessed.
+
+ - see: `ThreadConfined`
+ - see: `ThreadSafeReference`
+*/
+@propertyWrapper public class ThreadSafe<T: ThreadSafeWrappable> {
+    var threadSafeReference: ThreadSafeReference<T>?
+    var configuration: Realm.Configuration?
```

Comment:
Grabbing the RLMRealmConfiguration, opening a RLMRealm and then wrapping that in a Realm would eliminate a nontrivial bit of overhead.

Reply:
Does [`5004efb`](https://github.com/realm/realm-cocoa/pull/7402/commits/5004efb3144e73316e0a38d014a6f42407d0038d) do what you mean? And if so, what is the effect?
I was mainly looking at `RLMGetThreadLocalCachedRealmForPath`, but Swift `Realm` opens also hit this path. 
Is the main benefit that `Realm(_ rlmRealm: RLMRealm)` is one less call than `Realm(configuration: Configuration, queue: DispatchQueue? = nil)`?
Or is it significant that `Realm(_ rlmRealm:)` doesn't throw?

Reply:
The main benefit is that converting Realm.Configuration to RLMRealmConfiguration does a decent amount of work so it's better to do that only once.